debug_reg_printer.c

```diff
@@ -65,12 +68,15 @@ static unsigned int riscv_debug_reg_fields_to_s(char *buf, unsigned int offset,
 	riscv_debug_reg_ctx_t context, uint64_t value)
 {
 	unsigned int curr = offset;
-	curr += get_len_or_sprintf(buf, curr, " { ");
+	curr += get_len_or_sprintf(buf, curr, " {");
 	for (struct riscv_debug_reg_field_list_t list; get_next; get_next = list.get_next) {
 		list = get_next(context);
-		curr += riscv_debug_reg_field_to_s(buf, curr, list.field, context,
+		unsigned int len = riscv_debug_reg_field_to_s(buf, curr, list.field, context,
 				riscv_debug_reg_field_value(list.field, value));
-		curr += get_len_or_sprintf(buf, curr, ", ");
+		if (len) {
+			curr += len;
+			curr += get_len_or_sprintf(buf, curr, " ");
```

Comment:
Wouldn't you also like to skip the last separator (the one before the last field)? If so, I would suggest to move the check out of `riscv_debug_reg_field_to_s()` and rewrite the loop like so:
```
        const char *separator = "";
        for (struct riscv_debug_reg_field_list_t list; get_next; get_next = list.get_next) {
                list = get_next(context);
                if (<condition to skip the field>)
                        continue;
                curr += get_len_or_sprintf(buf, curr, separator);
                curr += riscv_debug_reg_field_to_s(buf, curr, list.field,
                                context, field_value);
                separator = ", ";
        }
```

Reply:
> I would suggest to leave the comma between the fields, since some value names include a space e.g. not supported.

Good point. I prefer to solve that by replacing whitespace in value names with _.

Reply:
> Wouldn't you also like to skip the last separator (the one before the last field)?

Yes, but I thought that was a pre-existing issue and it didn't bother me that much. I'll address it.

Reply:
Thanks!